pkg/gather/clusterconfig/operators.go

```diff
@@ -104,16 +105,114 @@ func gatherClusterOperators(ctx context.Context, configClient configv1client.Con
+	clogs := gatherPodContainersLogs(ctx, coreClient, pods, totalContainers)
+	if len(clogs) > 0 {
+		records = append(records, clogs...)
+	}
+
+	return records, nil
+}
+
+// gatherPodContainersLogs collect the pod current and previous containers logs
+func gatherPodContainersLogs(ctx context.Context, client corev1client.CoreV1Interface, pods []*corev1.Pod, total int) []record.Record {
+	var records []record.Record
+
+	// Fetch a list of containers in pods and calculate a log size quota
+	// Total log size must not exceed maxLogsSize multiplied by logCompressionRatio
+	klog.V(2).Infof("Found %d pods", len(pods))
+	bufferSize := int64(recorder.MaxLogSize * logCompressionRatio / total / 2)
+	klog.V(2).Infof("Maximum buffer size: %v bytes", bufferSize)
+	buf := bytes.NewBuffer(make([]byte, 0, bufferSize))
+
+	// Fetch previous and current container logs
+	for _, isPrevious := range []bool{true, false} {
+		for _, pod := range pods {
+			clog := collectPodContainerLogs(ctx, client, pod, isPrevious, buf, bufferSize)
+			if len(clog) > 0 {
+				records = append(records, clog...)
+			}
+		}
+	}
+
+	return records
+}
+
+func isPodRestarted(pod *corev1.Pod) bool {
+	// pods that have containers that have terminated with non-zero exit codes are considered failure
+	for _, status := range pod.Status.InitContainerStatuses {
+		if status.RestartCount > 0 {
+			return true
+		}
+	}
+	for _, status := range pod.Status.ContainerStatuses {
+		if status.RestartCount > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func collectPodContainerLogs(ctx context.Context, client corev1client.CoreV1Interface, pod *corev1.Pod, isPrevious bool, buf *bytes.Buffer, bufferSize int64) []record.Record {
+	var records []record.Record
+
+	allContainers := append(pod.Spec.InitContainers, pod.Spec.Containers...)
+	for _, c := range allContainers {
+		// only grab previous log if the pod is restarted
+		if isPrevious && !isPodRestarted(pod) {
+			continue
+		}
+
+		logName := fmt.Sprintf("%s_current.log", c.Name)
+		if isPrevious {
+			logName = fmt.Sprintf("%s_previous.log", c.Name)
+		}
+
+		buf.Reset()
+		klog.V(2).Infof("Fetching logs for %s container %s pod in namespace %s (previous: %v).", c.Name, pod.Name, pod.Namespace, isPrevious)
+		// Fetch container logs and continue on error
+		err := fetchPodContainerLog(ctx, client, pod, buf, c.Name, isPrevious, &bufferSize)
+		if err != nil {
+			klog.V(2).Infof("Error: %q", err)
+			continue
+		}
+
+		// Do not record empty records
+		if buf.Len() == 0 {
+			continue
+		}
+		records = append(records, record.Record{Name: fmt.Sprintf("config/pod/%s/logs/%s/%s", pod.Namespace, pod.Name, logName), Item: marshal.Raw{Str: buf.String()}})
+	}
+
+	return records
+}
+
+// clusterOperatorsReports generates the cluster operator reports
+func clusterOperatorsReports(ctx context.Context, items []configv1.ClusterOperator, dynamicClient dynamic.Interface, discoveryClient discovery.DiscoveryInterface) []record.Record {
```

Comment:
`clusterOperatorsReports` ? Didn't you want to call it `clusterOperatorsRecords`?